### PR TITLE
Reformat source to npm.im/standard format

### PR DESCRIPTION
**Pull request by jden**
This PR reformats radar and test code to conform to [`standard`](http://standardjs.com) style. From their homepage:

> No decisions to make. No .eslintrc, .jshintrc, or .jscsrc files to manage. It just works.

> This module saves you (and others!) time in two ways:

> No configuration. The easiest way to enforce consistent style in your project. Just drop it in.

> Catch style errors before they're submitted in PRs. Saves precious code review time by eliminating back-and-forth between maintainer and contributor.

It adds the `standard` cli tool to the npm `pretest` script so it will run locally and on ci.

Required
======
- [ ] +1 from team

Risks
=====
- Medium: This PR affects a significant percentage of the lines of code in the project, mostly with small, mechanical transformations via `standard --format`. The existing test suite passes and additional manual testing verifies this. The changes individually are all very low risk.

### CODE REVIEW

**Comment by jden** [comment]
As a process improvement, `standard` would have caught the undefined variable reference in #196

**Comment by jden** [comment]
Process to generate this PR:
1. `standard --format` to auto-apply as much formatting as possible
2. manually correct remaining issues (vast majority of which were single-var-blocks and whitespace)
3. ensure all tests pass

**Comment by jden** [comment]
I have rebased this branch against master to finish the format conversion to `standard`. This PR also adds running `standard` as a pretest step. It is ready for review and merging. Based on our conversation yesterday I'd like to get this merged to prevent creating more difficult merges for future PRs, because this PR is so far-reaching. Thanks!

File: api/api.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-  RadarApi = require('./apis/radar.js')
+var hostname = require('os').hostname()
+var Router = require('./lib/router.js')
+var RadarApi = require('./apis/radar.js')
 
 var api = new Router()
 
 function homepage (req, res) {
-  res.setHeader('Content-Type', 'text/plain'); // IE will otherwise try to save the response instead of just showing it.
+  res.setHeader('Content-Type', 'text/plain') // IE will otherwise try to save the response instead of just showing it
```
Comment (vanchi-zendesk) [inline api/api.js]: Removed the full stop from the comment. :D why? Is standard an enemy of good English? :D

Comment (vanchi-zendesk) [inline api/api.js]: /joke.

File: api/apis/radar.js [hunk 2]
```diff
@@ -126,27 +126,24 @@ function getPresence (req, res) {
       res.setHeader('Content-type', 'application/json')
       res.setHeader('Cache-Control', 'no-cache')
       res.setHeader('X-Radar-Host', hostname)
-
-
-
-      if (versionNumber === 2) {
+      if (versionNumber === 2) { // eslint-disable-line eqeqeq
```
Comment (vanchi-zendesk) [inline api/apis/radar.js]: no comment needed.

File: api/apis/radar.js [hunk 3]
```diff
@@ -126,27 +126,24 @@ function getPresence (req, res) {
...
   } else {
-    var scopes = q.scopes.split(','),
-      result = {}; // key: scope - value: replies
+    var scopes = q.scopes.split(',')
+    var result = {} // key: scope - value: replies
     scopes.forEach(function (scope) {
       var monitor = new PresenceManager('presence:/' + q.accountName + '/' + scope, {}, Presence.sentry)
       monitor.fullRead(function (online) {
-        if (versionNumber === 2) {
+        if (versionNumber === 2) { // eslint-disable-line eqeqeq
```
Comment (vanchi-zendesk) [inline api/apis/radar.js]: here as well.

File: api/lib/client.js [hunk 4]
```diff
@@ -142,18 +141,16 @@ Client.prototype._error = function (error, resData, callback) {
...
 
   // Parse location to check for port
-  parts = urlmodule.parse(response.headers.location)
-  if (parts.protocol == 'http:') {
-    options.secure = false
-    options.port = parts.port || 80
+  var parts = urlmodule.parse(response.headers.location)
+  if (parts.protocol === 'http:') {
+    this.options.secure = false
+    this.options.port = parts.port || 80
```
Comment (vanchi-zendesk) [inline api/lib/client.js]: TODO: maybe we ought to remove this entire section, since it's just not well tested.

File: configurator.js [hunk 5]
```diff
@@ -125,9 +125,9 @@ Configurator.prototype._defaultConfiguration = function () {
 }
 
 Configurator.prototype._pickFirst = function (propName) {
-  var values = [].slice.call(arguments, 1),
-    i = 0,
-    value
+  var values = [].slice.call(arguments, 1)
+  var i = 0
+  var value
```
Comment (vanchi-zendesk) [inline configurator.js]: initialize to false to be clearer.

File: api/api.js (hunk 1 above)
Comment (jden) [inline api/api.js]: oh - it was complaining about tailing whitespace after the full stop, and I must have just hit backspace an extra time :)

File: configurator.js (hunk 5 above)
Comment (vanchi-zendesk) [inline configurator.js]: hm.. maybe not. weird.

File: api/lib/client.js (hunk 4 above)
Comment (jden) [inline api/lib/client.js]: :+1: for completely deprecating and removing this part of the api- in a different PR ;)

File: configurator.js (hunk 5 above)
Comment (jden) [inline configurator.js]: I'm all for being explicit. I updated the PR to initialize it to `null` to indicate the absence of a value, and is also false-y.

File: core/lib/resources/presence/index.js [hunk 6]
```diff
@@ -175,7 +179,7 @@ Presence.prototype.unsubscribe = function (socket, message) {
 Presence.prototype.sync = function (socket, message) {
   var self = this
   this.fullRead(function (online) {
-    if (message.options && message.options.version == 2) {
+    if (message.options && message.options.version === 2) {
```
Comment (vanchi-zendesk) [inline core/lib/resources/presence/index.js]: same problems as before. This could come in as a string and it should still be fine.

Comment (jden) [inline core/lib/resources/presence/index.js]: good catch! I'll add a test for this case and fix

File: core/lib/resources/resource.js [hunk 7]
```diff
@@ -134,7 +133,7 @@ Resource.prototype.handleMessage = function (socket, message) {
 Resource.prototype.destroy = function () {}
 
 Resource.setBackend = function (backend) {
-  Persistence = backend
+  // Persistence = backend
```
Comment (vanchi-zendesk) [inline core/lib/resources/resource.js]: ??

Comment (vanchi-zendesk) [inline core/lib/resources/resource.js]: if we never use this, let's remove the method.

File: sample/views.js [hunk 8]
```diff
@@ -51,29 +51,29 @@ MessageList.prototype.render = function () {
...
-    var view = window.view[name]
-    view.render()
-  })
-}
+// function redraw () {
+//   Object.keys(view).forEach(function (name) {
+//     var view = window.view[name]
+//     view.render()
+//   })
+// }
```
Comment (vanchi-zendesk) [inline sample/views.js]: remove if not used.

File: server/server.js [hunk 9]
```diff
@@ -331,7 +329,7 @@ Server.prototype._sendErrorMessage = function (socket, value, origin) {
 
 // Initialize the current client
 Server.prototype._initClient = function (socket, message) {
-  var client = Client.create(message)
+  // do nothing
```
Comment (vanchi-zendesk) [inline server/server.js]: I think we still need to do Client.create(message)

File: tests/client.presence.remote.test.js [hunk 10]
```diff
@@ -82,6 +83,7 @@ describe('given a client and a server,', function () {
 
         p.fail_on_more_than(2)
         p.on(2, function () {
+          console.log('on 2')
```
Comment (vanchi-zendesk) [inline tests/client.presence.remote.test.js]: remove.

File: server/server.js (hunk 9 above)
Comment (jden) [inline server/server.js]: addressed in 3025143

File: tests/client.presence.remote.test.js (hunk 10 above)
Comment (jden) [inline tests/client.presence.remote.test.js]: addressed in 73f9c30

File: core/lib/resources/resource.js (hunk 7 above)
Comment (jden) [inline core/lib/resources/resource.js]: it looks like it's part of an abstract interface used for all resources (MessageList, etc). I'll comment it as a no-op, but if we want to change that interface it is out of scope for this pr

**Comment by jden** [comment]
I think I addressed all comments. Let me know and I'll squash & merge.

**Comment by vanchi-zendesk** [comment]
:+1:

**Pull request merged**